src/Illuminate/Database/Eloquent/Casts/AsArrayObject.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Illuminate\Database\Eloquent\Casts;
+
+use Illuminate\Contracts\Database\Eloquent\Castable;
+use Illuminate\Contracts\Database\Eloquent\CastsAttributes;
+
+class AsArrayObject implements Castable
+{
+    /**
+     * Get the caster class to use when casting from / to this cast target.
+     *
+     * @param  array  $arguments
+     * @return object|string
+     */
+    public static function castUsing(array $arguments)
+    {
+        return new class implements CastsAttributes {
+            public function get($model, $key, $value, $attributes)
+            {
+                return new ArrayObject(json_decode($attributes[$key], true));
```

Comment:
`ArrayObject` construct will only take an array, or object. `json_decode()` can possibly return `bool`, `null` among others.

(also worth noting `json_decode()` does not handle invalid json, which you either have to use the exception flag for, or `json_last_error()`)

https://www.php.net/manual/en/function.json-decode.php#refsect1-function.json-decode-returnvalues

Reply:
Then PHP will throw an error if it is not an error - which is what I would want to happen in this case. The column assigned this cast should always contain something that can be decoded to an array.

Reply:
That is bad. Throws exception web DB item is nullable (and is null).